Xamarin.Forms.Platform.WPF/Renderers/ImageRenderer.cs

```diff
@@ -141,19 +141,19 @@ public sealed class StreamImageSourceHandler : IImageSourceHandler
 	{
 		public async Task<System.Windows.Media.ImageSource> LoadImageAsync(ImageSource imagesource, CancellationToken cancelationToken = new CancellationToken())
 		{
-			BitmapImage bitmapimage = null;
-			StreamImageSource streamsource = imagesource as StreamImageSource;
-			if (streamsource != null && streamsource.Stream != null)
+			BitmapImage bitmapImage = null;
+
+			if (imagesource is StreamImageSource streamsource && streamsource.Stream != null)
 			{
-				using (Stream stream = await ((IStreamImageSource)streamsource).GetStreamAsync(cancelationToken))
-				{
-					bitmapimage = new BitmapImage()
-					{
-						StreamSource = stream
-					};
-				}
+				Stream stream = await ((IStreamImageSource)streamsource).GetStreamAsync(cancelationToken);
```

Comment:
why do you remove disposing stream. do you need it anywhere else?

Reply:
Agreed. Using `BitmapCacheOption.OnLoad` is correct, but then the `Stream` should be disposed. That's the point of using that option - so that you can dispose the stream after the bitmap is created.

Reply:
Yes, modified. Thanks!